addons/sale_timesheet/models/project_task.py

```diff
@@ -57,18 +58,31 @@ def _compute_remaining_hours_so(self):
     def _search_remaining_hours_so(self, operator, value):
         return [('sale_line_id.remaining_hours', operator, value)]
 
+    def _compute_last_sol_of_customer(self):
+        sol_per_domain = dict()
+        for task in self:
+            domain = tuple(task._get_last_sol_of_customer_domain())
+            if not domain:
+                task.last_sol_of_customer = False
+                continue
+            sol = sol_per_domain.get(domain)
+            if not sol:
+                sol = self.env['sale.order.line'].search(domain, limit=1)
+                sol_per_domain[domain] = sol
+            task.last_sol_of_customer = sol
+
     def _inverse_partner_id(self):
         super()._inverse_partner_id()
         for task in self:
             if task.allow_billable and not task.sale_line_id:
-                task.sale_line_id = task._get_last_sol_of_customer()
+                task.sale_line_id = task.last_sol_of_customer
 
     @api.depends('sale_line_id.order_partner_id', 'parent_id.sale_line_id', 'project_id.sale_line_id', 'allow_billable')
     def _compute_sale_line(self):
         super()._compute_sale_line()
         for task in self:
             if task.allow_billable and not task.sale_line_id:
-                task.sale_line_id = task._get_last_sol_of_customer()
+                task.sale_line_id = task.last_sol_of_customer
```

Comment:
Not sure if it is worth it, but it's maybe a good idea to bypass the ORM limit of 1000 (we never call a compute with a len(self) > 1000) to batch the compute on a bigger record set and have a better hit rate (if the performance improvements is still not enough).

Reply:
I guess for the moment, it is okay since we will rarely have more than 1000 records to recompute the sale_line_id but thanks for the idea. 🙂 

(By curiosity) how do we bypass the ORM limit of 1000? 🧐 🙂

Reply:
Something like that is possible:
```
huge_recordset._fields[<field_name>].compute_value(huge_recordset)
```
But I don't encourage you to do so because it will force the compute on all the recordset even if they have already be computed (and in cache then). It's still a hack.

Reply:
Okayyy thanks, and yes obviously, we will not do it here since it is not really useful right now, I was just curious to know how we could do that. 🙂